config/env.config.js

```diff
@@ -69,26 +71,38 @@ const RedirectPlugin = function(options) {
                             if (match && match.length > 1) {
                                 return match[1];
                             }
-                            return "";
+                            return null;
                         };
-                        var originUrl = decodeURIComponent(getParameterByName("originEnvironment"));
-                        var newUrl = window.localStorage.getItem("environmentConfig");
-                        if (newUrl && !window.location.href.match(newUrl) && originUrl !== newUrl) {
+                        var originUrl = getParameterByName("originEnvironment");
+                        var newUrl = getParameterByName("newEnvironment");
+                        if (newUrl) {
+                            newUrl = decodeURIComponent(newUrl);
+                            if (window.location.href.match(newUrl)) {
+                                window.localStorage.removeItem("environmentConfig");
```

Comment:
Oh, is it for the line `if (window.localStorage.getItem("environmentConfig")) {` below?  Would it make sense to just instead exit the function?  If you describe the flow in a comment (is it only the original manifest-pointed URL that needs to have the redirect config, or do they all?...) that would help.